src/node.ts

```diff
@@ -250,11 +250,26 @@ export class Link extends EventEmitter {
   // Number of children.
   length: number = 0;
 
+  name: string;
+
+  get steps() {
+    return this._steps;
+  }
+
+  // Recursively sync children steps, e.g. in case of dir rename
+  set steps(val) {
+    this._steps = val;
+    Object.entries(this.children).forEach(([, child]) => {
```

Comment:
```suggestion
    Object.values(this.children).forEach((child) => {
```

nit: I find `for...of` tends to be nicer than `forEach`, but that's not a blocker.

Reply:
Adjusted